#!/usr/bin/env python
PACKAGE = "rmagine_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

M_PI = 3.14159265359

gen.add("range_min", double_t, 0, "Minimum range", 0.0, 0.0, 10.0)
gen.add("range_max", double_t, 0, "Maximum range", 100.0, 0.0, 1000.0)

comp_enum = gen.enum([ gen.const("CPU",      int_t, 0, "CPU (Embree)"),
                       gen.const("GPU",     int_t, 1, "GPU (OptiX)")],
                     "Computing unit")

gen.add("computing_unit", int_t, 0, "Computing Unit", 0, 0, 1, edit_method=comp_enum)

gen.add("noise", bool_t, 0, "Enable noise", False)
gen.add("noise_mean", double_t, 0, "Noise mean", 0.0, -0.2, 0.2)
gen.add("noise_stddev", double_t, 0, "Noise standard deviation", 0.01, 0.0, 0.5)

exit(gen.generate(PACKAGE, "o1dn_simulator", "O1DnModel"))